src/app/shared/doc-viewer/header-link.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+
+/**
+ * Header link is a component that handles normalizing
+ * the anchor jump tags with the current route url.
+ *
+ * For example:
+ *
+ *    <a href="#foo">Foo</a>
+ *
+ * would result in the wrong url, this component
+ * combines the current route with that jump link:
+ *
+ *    <a href="/guide#foo">Foo</a>
+ */
+@Component({
+  selector: 'header-link',
+  template: `
+    <a
```

Comment:
I think this anchor would also benefit from an `aria-describedby` pointing to the header